### ISSUE TITLE
Does Mendeleev support SQLAlchemy=2.0.0 ?

### ISSUE THREAD

**Issue by jan-janssen** [lmmentel/mendeleev#95]
As `sqlalchemy==2.0.0` was recently released I was wondering if it is possible to support it. Currently the `pypi` package is restricted to `SQLAlchemy>=1.4.0,<2.0.0`.

**Comment by lmmentel** [comment]
Good point, I haven't checked what are the breaking changes in 2.0.0 but I'll loosen up the constraint and run the tests. If that works I think it's safe to bump it. WDYT?

**Comment by jan-janssen** [comment]
> Good point, I haven't checked what are the breaking changes in 2.0.0 but I'll loosen up the constraint and run the tests. If that works I think it's safe to bump it. WDYT?

Sounds good to me. While I am not an expert I just updated the `pyiron` code to `sqlalchemy=2.0.0`. For us it was a bit harder as we used the outdated database interface, but as you are just reading from the database, I assume you should be fine. 

While you are still on `sqlalchemy=1.4` you can also set the environment variable `SQLALCHEMY_WARN_20=1` this already raises warnings when you use outdated syntax. For more details on the migration https://docs.sqlalchemy.org/en/14/changelog/migration_20.html

**Comment by lmmentel** [comment]
I run the tests and there are some errors when fetching full tables. I'll look at it and will try to add a PR.

**Comment by lmmentel** [comment]
I narrowed down the issue to `read_sql` not being compatible with sqlalchemy 2.0.0 yet, more details https://github.com/pandas-dev/pandas/issues/40686. I'll wait for the fix on pandas side.

**Comment by jan-janssen** [comment]
> I narrowed down the issue to `read_sql` not being compatible with sqlalchemy 2.0.0 yet, more details [pandas-dev/pandas#40686](https://github.com/pandas-dev/pandas/issues/40686). I'll wait for the fix on pandas side.

Ok, makes sense - it seems to be a major change and several packages still need to update their interfaces.

**Comment by jan-janssen** [comment]
As `pandas` 2.0 was just released, I was wondering if it is now possible to update `mendeleev` as well.

**Comment by lmmentel** [comment]
Good point, I'll relax the version constraint and try to run the tests.
